survival-analysis/package_changes.Rmd

```diff
@@ -6,11 +6,11 @@ knitr::opts_chunk$set(echo = TRUE)
 
 These are notes from a session where we discussed the required changes to tidymodels. Some open issues are:
 
-* Where does the user specify the time points for certain performance metrics (e.g. ROC methods)? There could be an option to the `tune` functions (or their respective control functions) for this. Alternatively, `.time` could be encoded into a metric set. In either case, the `.time` vector will be passed to the `parsnip` predict calls (when appropriate) so that the right predictions are created. We will _not_ have a `.time` argument to `yardstick` metrics since the times will be in the results of the prediction functions. 
+* Where does the user specify the time points for certain performance metrics (e.g. ROC methods)? There could be an option to the `tune` functions (or their respective control functions) for this. Alternatively, `.time` could be encoded into a metric set. In either case, the `.time` vector will be passed to the `parsnip` predict calls (when appropriate) so that the right predictions are created. We will _not_ have a `.time` argument to `yardstick` metrics since the times will be in the results of the prediction functions. [Note, the argument to `predict()` is called `time`, the column in the prediction object is called `.time`. In the rest of these notes, we use `.time` as the name for the vector of time at which we predict survival probability or hazard.]
 
 * `tune` has some functions that require the user to pick a metric. For example, `tune_bayes()` will optimize on the first metric in the metric set. Functions like `select_best()` or `autoplot()` have specific arguments for 1+ metrics. It is unclear how the user should pick a metric when some of them are computed at specific times. 
 
-* There was some discussion about packages like `workflows` (and maybe `recipes`) making the time/event columns into a `Surv` object. This would simplify the changes to `tune` and other packages but increase the complexity to `workflows`. 
+* There was some discussion about packages like `workflows` (and maybe `recipes`) making the time/event columns into a `Surv` object. This would simplify the changes to `tune` and other packages but increase the complexity to `workflows`. Our current thinking is to require the user to make a `Surv` object in a column in the data frame, prior to any analysis. This follows the recommendation to do any modifications of the response prior to recipes etc.
 
 * The possibility of adding a subclass to `model_fit` objects for these types of models. 
```

Comment:
Is this covered by having a dedicated mode?

Reply:
Yes. we can remove that